File: Languages/Golang/Account.go
package main

type Account struct {
	accountNumber    int
	pin              int
	avaliableBalance float64
	totalBalance     float64
}

func NewAccount(theAccountNumber int, thePin int, theAvailableBalance float64, theTotalBalance float64) *Account {
	return &Account{accountNumber: theAccountNumber, pin: thePin, avaliableBalance: theAvailableBalance, totalBalance: theTotalBalance}
}

// validate user pin
func (a *Account) validatePin(userPin int) bool {
	if userPin == a.pin {
		return true
	} else {
		return false
	}
}

// get the available balance
func (a *Account) getAvailableBalance() float64 {
	return a.avaliableBalance
}

// get the total balance
func (a *Account) getTotalBalance() float64 {
	return a.totalBalance
}

// credit an amount to the account
func (a *Account) credit(amount float64) {
	a.totalBalance += amount
}

// debit an amount from the account
func (a *Account) debit(amount float64) {
	a.avaliableBalance -= amount
	a.totalBalance -= amount
}

// get account number
func (a *Account) getAccountNumber() int {
	return a.accountNumber
}

File: Languages/Java/Account.java
package Languages.Java;

public class Account {
    private int accountNumber;
    private int pin; 
    private double availableBalance; 
    private double totalBalance; 


    public Account(int theAccountNumber, int thePin, double theAvailableBalance, double theTotalBalance) {
        accountNumber = theAccountNumber;
        pin = thePin;
        availableBalance = theAvailableBalance;
        totalBalance = theTotalBalance;
    }

    // validate user pin 
    public boolean validatePin(int userPin) {
        if (userPin == pin) {
            return true; 
            
        } else {
            return false; 
        }
    }

    // returns the available balance 
    public double getAvailableBalance() {
        return availableBalance; 
    }

    // returns the total balance 
    public double getTotalBalance() {
        return totalBalance; 
    }

    // credits an amount into account, need an amount and add that to total balancce 
    public void credit(double amount) {
        totalBalance += amount; 
    }


    // debits an amount from the account, need an amount and subtract that from total balance
    public void debit(double amount) {
        availableBalance -= amount; // subtract from available balance 
        totalBalance -= amount; // subtract from total balance
    }

    // get the users accounbt number 
    public int getAccountNumber() {
        return accountNumber; 
    }
    
}
